feat(kstepper): new component [khcp-4301] (#745)
Kong/kongponents: docs/components/stepper.md

# Stepper

> **KStepper** - A basic Stepper component

Reviewer: nit
```suggestion
**KStepper** - An ordered Stepper component
```

<KStepper :steps="defaultItems" />

```html
<KStepper :steps="steps" />
```

## Props

### steps

> An array of `KSteps`. Each `KStep` has a `label` and an optional `state`.

Reviewer: Since we don't expose the usage of the `KStep` component, I think we should update the references in this file to just talk about steps as objects within an array instead of referring to the internal-only component
```suggestion
An array of step objects. Each step object should have a required `label` property, and an optional `state` property.
```

<div>
  <KStepper :steps="[
      { label: 'Step  a long long long long time ago', state: 'completed' },
      { label: 'in a galaxy far far away', state: 'completed' },
      { label: 'Kongponents were battling in space and', state: 'pending' },
      { label: 'fighting robots and space monsters with lots of explosions' }
    ]"
  />
</div>

```html
<KStepper :steps="[
    { label: 'Step  a long long long long time ago', state: 'completed' },
    { label: 'in a galaxy far far away', state: 'completed' },
    { label: 'Kongponents were battling in space and', state: 'pending' },
    { label: 'fighting robots and space monsters with lots of explosions' }
  ]"
/>
```

### maxLabelWidth

> The width of step labels (default is `170`). Currently we support numbers (will be converted to `px`), `auto`, and percentages for values.

Reviewer: I know the default is actually a `number` of `170` but can we call out here in the docs that it's pixels just so there's no confusion?
```suggestion
The width of step labels (default is `170px`). We support numbers (will be converted to `px`), `auto`, and percentages (e.g. `25%`) for values.
```

<KStepper :steps="longSteps" max-label-width="100" />

```html
<KStepper :steps="steps" max-label-width="100" />
```

## KStep

> **KStepper** generates a **KStep** for each item in the `steps` property. Each step has a `label`, `state`, and an associated divider (unless it's the last step.)

Reviewer: Again since we don't expose the usage of the `KStep` component, I believe this header and sentence can go away.

Author: Fixed

### Properties

- `label` (required) - the text displayed beneath the step

> - `state` - the state of the step controls the icon, we support: `completed`, `pending`, and `error`

Reviewer: ```suggestion
- `state` - the state of the step controls the icon, we support: `completed`, `pending`, and `error`. If a 'state' is not provided, it will show the default grey icon.
```

> `completed` steps result in a colored following divider.

Reviewer: Just feel like this should be reworded
```suggestion
A step with a state of `completed` results in a filled-in divider.
```

> `completed` steps result in a colored following divider.

Reviewer: Can you move the properties under the `steps` prop documentation?

Author: Fixed

<div>
  <KStepper :steps="[
      { label: 'A completed step', state: 'completed' },
      { label: 'End' }
    ]"
  />
</div>

```javascript

> [{ label: 'A completed step', state: 'completed' },
>  { label: 'End' }]

Reviewer: Formatting
```suggestion
[
  { label: 'A completed step', state: 'completed' },
  { label: 'End' }
 ]
```

```

Both `pending` and `error` states will bold the label, because these 2 states indicate the "current" step. `completed` indicates past steps, while `default` indicates future steps.

<div>
  <KStepper :steps="[
      { label: 'A pending step', state: 'pending' },
      { label: 'End' }
    ]"
  />
</div>

> ```javascript
> [{ label: 'A pending step', state: 'pending' },
>  { label: 'End' }]
> ```

Reviewer: Formatting
```suggestion
```js
[
  { label: 'A pending step', state: 'pending' },
  { label: 'End' }
]
```
```

Author: Fixed

<div>
  <KStepper :steps="[
      { label: 'An erroneous step', state: 'error' },
      { label: 'End' }
    ]"
  />
</div>

> ```javascript
> [{ label: 'An erroneous step', state: 'error' },
>  { label: 'End' }]
> ```

Reviewer: Formatting 
```suggestion
```js
[
  { label: 'An erroneous step', state: 'error' },
  { label: 'End' }
]
```
```

Author: Fixed

The last step will never have a following divider.

<div>
  <KStepper :steps="[
      { label: 'A default step' }
    ]"
  />
</div>

> ```javascript

Reviewer: ```suggestion
```js
```

# Stepper

**KStepper** - An ordered Stepper component

<KStepper :steps="defaultItems" />

```html
<KStepper :steps="steps" />
```

## Props

### steps

An array of step objects. Each step object should have a required `label` property, and an optional `state` property.

<div>
  <KStepper :steps="[
      { label: 'Step  a long long long long time ago', state: 'completed' },
      { label: 'in a galaxy far far away', state: 'completed' },
      { label: 'Kongponents were battling in space and', state: 'pending' },
      { label: 'fighting robots and space monsters with lots of explosions' }
    ]"
  />
</div>

```html
<KStepper :steps="[
    { label: 'Step  a long long long long time ago', state: 'completed' },
    { label: 'in a galaxy far far away', state: 'completed' },
    { label: 'Kongponents were battling in space and', state: 'pending' },
    { label: 'fighting robots and space monsters with lots of explosions' }
  ]"
/>
```

#### Properties

- `label` (required) - the text displayed beneath the step

> - `state` - the state of the step controls the icon, we support: `completed`, `pending`, and `error`. If a 'state' is not provided, it will show the default grey icon.

Author: ```suggestion
- `state` - the state of the step controls the icon, we support: `completed`, `pending`, and `error`. If a state property is not provided, it will show the default grey icon.
```

> A step with a state of `completed` results in a filled-in divider.

Author: ```suggestion
#### States

A step with a state of `completed` results in a filled-in divider.
```
